src/components/v2/V2Project/V2ProjectReconfigureModal/V2ReconfigureUpcomingMessage.tsx

```diff
@@ -0,0 +1,64 @@
+import { Trans } from '@lingui/macro'
+import { V2ProjectContext } from 'contexts/v2/projectContext'
+import { useContext } from 'react'
+import { BigNumber } from '@ethersproject/bignumber'
+
+import { detailedTimeString, secondsUntil } from 'utils/formatTime'
+
+import { getBallotStrategyByAddress } from 'constants/v2/ballotStrategies/getBallotStrategiesByAddress'
+
+export default function V2ReconfigureUpcomingMessage() {
+  const { fundingCycle } = useContext(V2ProjectContext)
+  if (!fundingCycle) return null
+
+  const currentFCNumber = fundingCycle.number.toNumber()
+
+  const ballotStrategy = getBallotStrategyByAddress(fundingCycle.ballot)
+  const ballotStrategyLength = ballotStrategy.length
```

Comment:
Note, we can actually do a contract read for the ballot duration, which we probably should do so that the contracts are the single source of truth.

No biggie for now.

Reply:
Got it